system/Database/MigrationRunner.php

```diff
@@ -839,7 +839,8 @@ protected function migrate($direction, $migration): bool
             throw new RuntimeException($message);
         }
 
-        $instance = new $class();
+        /** @var Migration $instance */
+        $instance = new $class(Database::factory()->loadForge($this->db));
```

Comment:
Why do you need to add Database::factory() method?

```suggestion
        $instance = new $class(Database::forge($this->db));
```

Reply:
Done.